spec/features/snacks/show_spec.rb

```diff
@@ -0,0 +1,84 @@
+    end
+
+    it "then I see the list of all locations with vending machines that carry that snack." do
+      owner = Owner.create!(name: "Sam's Snacks")
+      dons  = owner.machines.create!(location: "Don's Mixed Drinks")
+      dans  = owner.machines.create!(location: "Dan's Vending Machines")
+      cranapple = Snack.create!(name: "Cranapple", price: 1.00)
+      crangrape = Snack.create!(name: "Crangrape", price: 1.50)
+      cranpineapple = Snack.create!(name: "Cranpineapple", price: 2.00)
+      MachineSnack.create!(machine: dons, snack: cranapple)
+      MachineSnack.create!(machine: dons, snack: crangrape)
+      MachineSnack.create!(machine: dans, snack: crangrape)
+      MachineSnack.create!(machine: dons, snack: cranpineapple)
+
+      visit snack_path(crangrape)
+
+      expect(page).to have_content(dans.location)
+      expect(page).to have_content(dons.location)
+    end
+
+    it "then I see the average price for snacks in that vending machine." do
+      owner = Owner.create!(name: "Sam's Snacks")
+      dons  = owner.machines.create!(location: "Don's Mixed Drinks")
+      dans  = owner.machines.create!(location: "Dan's Vending Machines")
+      cranapple = Snack.create!(name: "Cranapple", price: 1.00)
+      crangrape = Snack.create!(name: "Crangrape", price: 1.50)
+      cranpineapple = Snack.create!(name: "Cranpineapple", price: 2.00)
+      MachineSnack.create!(machine: dons, snack: cranapple)
+      MachineSnack.create!(machine: dons, snack: crangrape)
+      MachineSnack.create!(machine: dans, snack: crangrape)
+      MachineSnack.create!(machine: dons, snack: cranpineapple)
+
+      visit snack_path(crangrape)
+      within "#machine-#{dons.id}" do 
+        expect(page).to have_content("Average Snack Price: $1.50")
+      end
+
+      within "#machine-#{dans.id}" do 
+        expect(page).to have_content("Average Snack Price: $1.50")
+      end
+    end
+
+    it "then I see the count for all different kinds of snacks in that vending machine." do 
+      owner = Owner.create!(name: "Sam's Snacks")
+      dons  = owner.machines.create!(location: "Don's Mixed Drinks")
+      dans  = owner.machines.create!(location: "Dan's Vending Machines")
+      cranapple = Snack.create!(name: "Cranapple", price: 1.00)
+      crangrape = Snack.create!(name: "Crangrape", price: 1.50)
+      cranpineapple = Snack.create!(name: "Cranpineapple", price: 2.00)
+      MachineSnack.create!(machine: dons, snack: cranapple)
+      MachineSnack.create!(machine: dons, snack: crangrape)
+      MachineSnack.create!(machine: dans, snack: crangrape)
+      MachineSnack.create!(machine: dons, snack: cranpineapple)
+
+      visit snack_path(crangrape)
+      
+      within "#machine-#{dons.id}" do 
+        expect(page).to have_content("Number of Snacks in Machine: 3")
+      end
+
+      within "#machine-#{dans.id}" do 
+        expect(page).to have_content("Number of Snacks in Machine: 1")
+      end
+    end
```

Comment:
I think you could make this easier for yourself, and just make the assertion that the count and the average is within that specific part of the page in just one within. AKA, make one test out of these two it blocks, and assert both number of snacks and average snack price for each within block